File: leetcode/136.Single_Number/py3/sol.py
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        theSet = set()
        for it in nums:
          if it in theSet:
            theSet.remove(it)
          else:
            theSet.add(it)

        print('the Set: ', theSet)
        ans = theSet.pop()
        return ans

sol = Solution()
inp = [2,2,1,1,3,3,4]
ret = sol.singleNumber(inp)
print('ret: ', ret)

File: leetcode/136.Single_Number/js/sol.js


/**
 * @param {number[]} nums
 * @return {number}
 */
 var singleNumber = function(nums) {
  let theSet = new Set()
  for(let it = 0; it < nums.length; it++) {
    if (!theSet.has(nums[it]) ) {
      theSet.add(nums[it])
    } else {
      theSet.delete(nums[it])
    }
  }
  let ans
  for (let it of theSet) {
      ans = it
  }
  console.log('ans: ', ans)
  return ans
};

// inp = [1, 2, 2]
// ret = singleNumber(inp)
// console.log('ret: ', ret)
